jaeger-core/src/main/java/io/jaegertracing/internal/JaegerSpanContext.java

```diff
@@ -122,11 +126,17 @@ public static JaegerSpanContext contextFromString(String value)
       oibe: because java doesn't like to convert large hex strings to longs
       we should write this manually instead of using BigInteger.
     */
-    return new JaegerSpanContext(
+    return factory.createSpanContext(
         new BigInteger(parts[0], 16).longValue(),
         new BigInteger(parts[1], 16).longValue(),
         new BigInteger(parts[2], 16).longValue(),
-        new BigInteger(parts[3], 16).byteValue());
+        new BigInteger(parts[3], 16).byteValue(),
+        Collections.<String, String>emptyMap(),
+        null);
+  }
+
+  protected TracingFactory tracingFactory() {
+    return new TracingFactory();
```

Comment:
factory should be instantiated only once

Reply:
I was thinking about using a singleton. I guess I should.

Reply:
not a singleton in the sense of a private static member, but just an externally provided object.